lnwallet/channel.go

```diff
@@ -5935,8 +6150,13 @@ func (lc *LightningChannel) ReceiveHTLC(htlc *lnwire.UpdateAddHTLC) (uint64, err
 
 	// Clamp down on the number of HTLC's we can receive by checking the
 	// commitment sanity.
+	// We are not enforcing the FeeBuffer here because our peer is not
+	// aware of it and neither should they be. This is one of the reasons
+	// the FeeBuffer was introduced so we use it and therefore do not
+	// enforce the additional buffer.
```

Comment:
Rephrased the comment, so the main reason we do not use it here is for the concurrency problem we try to solve:

```
	// We do not enforce the FeeBuffer here because one of the reasons it
	// was introduced is to protect against asynchronous sending of htlcs so
	// we use it here. The current lightning protocol does not allow to
	// reject ADDs already sent by the peer.
```

Reply:
The point you are making, that we should look at whether the htlc ADDs balance to our local will be implemented in a future PR, this relates to the "allow dipping the peer below the reserve" question.